// codespaces/dist/utils/transform.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getCodespacesEnvironment = void 0;
const environment_file_1 = require("../utils/environment-file");
const getCodespacesEnvironment = () => {
    const contents = (0, environment_file_1.getEnvironmentFileContents)();
    if (!contents)
        return {
            isCodespaces: false,
            session: {},
            git: { user: {}, repository: {} },
        };
    return {
        isCodespaces: true,
        name: contents.CODESPACE_NAME,
        branch: contents.Branch,
        owner: contents.Owner,
        version: contents.ContainerVersion,
        session: {
            id: contents.SESSION_ID,
        },
        git: {
            user: {
                name: contents.GIT_CONFIG_USER_NAME,
                email: contents.GIT_CONFIG_USER_EMAIL,
            },
            repository: {
                name: contents.RepoName,
                url: contents.GIT_REPO_URL,
                branch: contents.GIT_REPO_BRANCH_NAME,
            },
        },
    };
};
exports.getCodespacesEnvironment = getCodespacesEnvironment;
